### ISSUE TITLE
Tighten /var/log/ default file permissions from 644 to 640

### ISSUE THREAD

**Issue by candrews** [shadow-maint/shadow#679]
I'm working on evaluating Debian against STIGs and CIS benchmarks and one of the findings reported is:

Verify permissions of log files: http://static.open-scap.org/ssg-guides/ssg-ubuntu2004-guide-stig.html#xccdf_org.ssgproject.content_rule_permissions_local_var_log

This rule ensure that files in /var/log have 640 permissions.

This project seems to create/own a number of the files in /var/log and it sets the file permissions to 644.

640 makes more sense to me - there doesn't seem to be any reason for a regular user to read these logs.

Could you please consider using the more restrictive 640 permissions for the /var/log/ files, improving security by default?

**Comment by hallyn** [comment]
Which specific logfiles would you suggest changing?

Generally speaking, that means that you'll have to become root (or syslog) to read the logs.  Training people to do more as root is its own problem.

**Comment by candrews** [comment]
> Which specific logfiles would you suggest changing?

The rule (which is part of the CIS benchmarks and STIG) states that everything in `/var/log` must have 640 permissions: https://github.com/ComplianceAsCode/content/blob/v0.1.66/linux_os/guide/system/permissions/files/permissions_local_var_log/rule.yml#L56

Therefore, in accordance with this rule, I'm suggesting that all files in `/var/log` provided by shadow be changed to 640 permissions.

The files in /var/log present in a very basic Debian installation are:
```
$ docker run -it debian:sid ls -la /var/log
total 0
drwxr-xr-x. 1 root root 50 Feb 27 00:00 .
drwxr-xr-x. 1 root root 90 Feb 27 00:00 ..
drwxr-xr-x. 1 root root  0 Feb  8 16:07 apt
-rw-rw----. 1 root utmp  0 Feb 27 00:00 btmp
-rw-r--r--. 1 root root  0 Feb 27 00:00 faillog
-rw-rw-r--. 1 root utmp  0 Feb 27 00:00 lastlog
-rw-rw-r--. 1 root utmp  0 Feb 27 00:00 wtmp
```

I don't know which of those files are managed by shadow and I'm having a rough time figuring that out. If you know how to tell which packages create those files, can you please share that info?

> Generally speaking, that means that you'll have to become root (or syslog) to read the logs. Training people to do more as root is its own problem.

I agree that training people to do more as root is problematic. But these logs do contain sensitive information so they really shouldn't be readable by just anyone (as exposing this sensitive information is also problematic). The permission required by the rule is 640, so the files are group readable. Therefore, using 640, users in the `utmp` group could still read `/var/log/{lastlog,wtmp,btmp}`, so the special users who need to be able to read these logs could be added to that group.

**Comment by hallyn** [comment]
The project you linked to is out of date.

https://www.tenable.com/audits/items/DISA_STIG_SLES_15_v1r9.audit:c3ab5c3bf07887659067e3d5a3c784c0  provides an exception for btmp, wtmp, and lastlog.

**Comment by candrews** [comment]
> The project you linked to is out of date.
> 
> https://www.tenable.com/audits/items/DISA_STIG_SLES_15_v1r9.audit:c3ab5c3bf07887659067e3d5a3c784c0 provides an exception for btmp, wtmp, and lastlog.

I'm not so sure it's out of date. You linked to the information for SLES. The information for Ubuntu doesn't have those exceptions: https://www.tenable.com/audits/items/DISA_STIG_Ubuntu_20.04_LTS_v1r5.audit:a9e7d4a8941a4fee48cbec0b5e72484a

**Comment by candrews** [comment]
Why was this completed?

BTW, I emailed DISA at disa.stig_spt@mail.mil to ask why the SLES and Ubuntu STIGs are different. They didn't tell me why, but they did say that the difference is not a mistake. So I think this request is still valid.

**Comment by hallyn** [comment]
The issue was closed not as completed but as invalid.

Some of the reasons why it is invalid include:
1. Shadow does not create these files in the first place.
2. If it did, then we would absolutely not hard-code the 0640 permissions.  We'd work on flexible support using something like /etc/login.defs .
3. The sles link above shows that the spec is misguided.
4. The sles link above shows that distros are not agreed on how to handle this.

Lots of people have bad ideas.  If we jumped to action every time a
group documented theirs, we'd ... be even more swamped than we are.
